GRDB/Record/PersistableRecord+Encodable.swift

```diff
@@ -40,7 +42,23 @@ private struct PersistableRecordKeyedEncodingContainer<Key: CodingKey> : KeyedEn
             // This allows us to encode Date as String, for example.
             encode((value as! DatabaseValueConvertible), key.stringValue)
         } else {
-            try value.encode(to: PersistableRecordEncoder(codingPath: [key], encode: encode))
+            do {
+                 try value.encode(to: PersistableRecordEncoder(codingPath: [key], encode: encode))
+            } catch {
```

Comment:
Why do we carefully check for the error type before we switch to JSON **decoding**, but not when we switch to JSON **encoding**? I'd like some consistency, and the same level of care to be displayed in both places.

Reply:
This was a oversight, fixed now that you pointed it out.

Reply:
You mean in 6262d10794f08350ff284ae667f1fe2e60bd2c99? It still needs a little extra effort: just like we did for decoding, we have to check for a **particular** encoding error before switching to JSON.

Reply:
Updated in d81475ad